misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt

```diff
@@ -39,18 +66,51 @@ class FakeRedis : Redis {
       return null
     }
 
-    return value.value
+    return value.data
   }
 
-  // Sets the value for a key, the key does not expire
   override fun set(key: String, value: String): String {
     // Set the key to expire at the latest possible instant
-    keyValueStore[key] = Value(value = value, expiryInstant = Instant.MAX)
-    return value
+    keyValueStore[key] = Value(
+      data = value.toByteArray(Charsets.UTF_8),
+      expiryInstant = Instant.MAX
+    )
+
+    return StatusCode.OK.toString()
+  }
+
+  override fun set(key: String, value: ByteArray): String {
+    // Set the key to expire at the latest possible instant
+    keyValueStore[key] = Value(
+      data = value,
+      expiryInstant = Instant.MAX
+    )
+
+    return StatusCode.OK.toString()
   }
 
   override fun set(key: String, expiryDuration: Duration, value: String): String {
-    keyValueStore[key] = Value(value = value, expiryInstant = clock.instant().plusSeconds(expiryDuration.seconds))
-    return value
+    keyValueStore[key] = Value(
+      data = value.toByteArray(Charsets.UTF_8),
+      expiryInstant = clock.instant().plusSeconds(expiryDuration.seconds)
+    )
+
+    return StatusCode.OK.toString()
+  }
+
+  override fun set(key: String, expiryDuration: Duration, value: ByteArray): String {
+    keyValueStore[key] = Value(
+      data = value,
+      expiryInstant = clock.instant().plusSeconds(expiryDuration.seconds)
+    )
+
+    return StatusCode.OK.toString()
+  }
+
+  companion object {
+    // Status codes returned by redis operations
```

Comment:
kind of strange that we return a string "OK" for successful operations. Is that useful for clients? Are they expected to assert that the response of the operation `.equals("OK")`? What does RealRedis return?

My recommendation would be to expect success if an exception is not thrown.

Reply:
and perhaps return the set value?

Reply:
My reasoning here was to have `FakeRedis` behave like `RealRedis`. Maybe the correct thing to do here is to modify `RealRedis` to have this behaviour instead (return the set value instead of returning the redis response as a string).